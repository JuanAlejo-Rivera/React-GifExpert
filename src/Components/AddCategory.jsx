//Aqui creamos un input y emitimos un valor cuando el usuario da enter


import { useState } from "react"
import PropTypes from 'prop-types'

export const AddCategory = ({onNewCategory}) => {
  const [inputValue, setInputValue] = useState("")

  const onInputChange = ({ target }) => {
    setInputValue(target.value)
  }

  const onSubmit = (event) =>{
    event.preventDefault();
    if(inputValue.trim().length <= 1)return;
      // setCategories(ActCategories => [inputValue, ...ActCategories])
      //El ActCategories copia lo que esta en el setCategories actualmente esto debido al hook, por eso no se pasa por los props
      onNewCategory(inputValue.trim())
      //El valor de inputValue se envia a onNewCategory definida en el padre para ejecutar la funcion onAddCategory
      setInputValue('')
    
  }

  return (
    // <form onSubmit={onSubmit} aria-label="formulario de busqueda">
    <form onSubmit={onSubmit} aria-label="form">
      <input 
        type="text"
        placeholder="Buscar Gifs"
        value={inputValue}
        onChange={onInputChange}
      />
    </form>
  )
}


AddCategory.prototype ={
  onNewCategory : PropTypes.func.isRequired,
}