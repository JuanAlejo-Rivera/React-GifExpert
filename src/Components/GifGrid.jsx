//Aqui ejecuto la api de los gif y le agrego la category, llamo la funcion desde getGifs.js

import { GifItem } from "./GifItem"
import { useFetchGifs } from "../hooks/useFetchGifs";
import PropTypes from 'prop-types'

export const GifGrid = ({ categoryP }) => {
//Custom hook
    const {isLoading, images} = useFetchGifs(categoryP);


    return (
        <>
            <h3>{categoryP}</h3>
            {
                isLoading && (<h2>Cargando...</h2>)
            }
            
            {/* Creamos lista ordenada */}
            <div className="card-grid">
                {/* Mapeamos las imagenes, entre {} por que es la sintaxis de js en jsx la funcion inia con () por el retorno implicito*/}
                {images.map((image) =>(
                    <GifItem 
                    key={image.id}
                        {...image}//toma todas las propiedades del objeto image y las pasa al componente GifItem de forma desestructurada.
                    />
                ))}

            </div>
        </>
    )
}


GifGrid.propTypes = {
    categoryP: PropTypes.string.isRequired,
}