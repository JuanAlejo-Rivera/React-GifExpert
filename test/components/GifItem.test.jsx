import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/Components/GifItem";

describe('Prueba gifItem', () => {

  const title = 'sukuna'
  const url = 'https://sukuna.jpg'

  test('Debo hacer un snapchot', () => {

    const { container } = render(<GifItem title={title} url={url} />)
    expect(container).toMatchSnapshot();

  });

  test('Debe mostrar la imagen con el url y el alt indicado', () => {

    render(<GifItem title={title} url={url} />);
    //muestra una foto del codigo sirve para recordar que estoy evaluando
    // screen.debug();
    // expect(screen.getByRole('img').src).toBe(url);
    // expect(screen.getByRole('img').alt).toBe(title);

    const {src, alt} = screen.getByRole('img')
    expect(src).toBe(src)
    expect(alt).toBe(alt)
  })

  test('Debe de mostar el titulo en el componente', () => { 
    
    render(<GifItem title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy()

   })

});