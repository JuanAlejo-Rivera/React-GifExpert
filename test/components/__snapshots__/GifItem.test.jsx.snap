// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba gifItem Debo hacer un snapchot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="sukuna"
      src="https://sukuna.jpg"
    />
    <p>
      sukuna
    </p>
  </div>
</div>
`;
